File: packages/ui/src/tokens.ts
/**
 * FROZEN CONTRACT — the only source of visual values in the system. [INV-13]
 *
 * ESLint bans hex literals everywhere else. Additions here are allowed;
 * changing an existing token requires an ADR.
 */

export const colour = {
  // Operation colours — luminance-separated so they survive deuteranopia
  // and protanopia before shape coding is even applied.
  coral: '#FF6B5B',
  marine: '#2E8FD6',
  kelp: '#3FBF87',
  sunfish: '#FFC531',
  violet: '#9B6BE8',
  // Surfaces and text
  deep: '#12324F',
  sand: '#FFF6E9',
  foam: '#E6F4FB',
  tide: '#0E7C9B',
  // Parent zone — deliberately a different world. A child glancing at a parent
  // screen should instantly read "not mine".
  parentTeal: '#0F766E',
  parentSlate: '#334155',
  parentMist: '#F1F5F9',
} as const;

/** Colour carries the operation; shape and glyph carry it redundantly. */
export const operationColour = {
  add: colour.coral,
  sub: colour.marine,
  mul: colour.kelp,
  div: colour.sunfish,
  wild: colour.violet,
} as const;

export const operationShape = {
  add: 'roundedSquare',
  sub: 'circle',
  mul: 'hexagon',
  div: 'diamond',
  wild: 'star',
} as const;

export const operationGlyph = {
  add: '+',
  sub: '\u2212',
  mul: '\u00D7',
  div: '\u00F7',
  wild: '\u2726',
} as const;

/** Pattern coding used when colour-vision support is enabled. */
export const operationPattern = {
  add: 'dots',
  sub: 'waves',
  mul: 'crosshatch',
  div: 'diagonal',
  wild: 'burst',
} as const;

export const space = [0, 4, 8, 12, 16, 24, 32, 48, 64] as const;

export const radius = { tile: 8, card: 16, sheet: 24, pill: 999 } as const;

export const outline = { standard: 2, strong: 4, focus: 4 } as const;

export const opacity = { full: 1, disabled: 0.48 } as const;

export const icon = { small: 24, medium: 32, large: 48 } as const;

export const type = {
  /** Always the largest type in the app. Never demote it. */
  target: 56,
  tile: 34,
  preview: 28,
  h1: 24,
  body: 17,
  caption: 14,
} as const;

export const font = {
  display: 'Baloo2-ExtraBold',
  body: 'Nunito-SemiBold',
  bodyBold: 'Nunito-ExtraBold',
  dyslexic: 'Lexend-Regular',
} as const;

/** Hard floor for any interactive element in the child zone. [INV-14] */
export const touch = { min: 64 } as const;

export const motion = {
  fast: 120,
  base: 260,
  slow: 450,
  /** Every duration collapses to this when reduced motion is on. */
  reduced: 100,
  /** No animation may block input for longer than this. */
  maxInputBlockMs: 300,
} as const;

/** Text scaling values exercised by the mandatory accessibility fixtures. */
export const accessibility = {
  textScale: {
    base: 1,
    large: 1.3,
    extraLarge: 1.6,
  },
} as const;

export const tokens = {
  colour,
  operationColour,
  operationShape,
  operationGlyph,
  operationPattern,
  space,
  radius,
  outline,
  opacity,
  icon,
  type,
  font,
  touch,
  motion,
  accessibility,
} as const;

export type Tokens = typeof tokens;

File: flutter/packages/design_system/lib/src/tokens.dart
import 'package:flutter/material.dart';

/// Existing values mirror `packages/ui/src/tokens.ts` and are frozen.
abstract final class PokoColour {
  static const coral = Color(0xFFFF6B5B);
  static const marine = Color(0xFF2E8FD6);
  static const kelp = Color(0xFF3FBF87);
  static const sunfish = Color(0xFFFFC531);
  static const violet = Color(0xFF9B6BE8);
  static const deep = Color(0xFF12324F);
  static const sand = Color(0xFFFFF6E9);
  static const foam = Color(0xFFE6F4FB);
  static const tide = Color(0xFF0E7C9B);
  static const parentTeal = Color(0xFF0F766E);
  static const parentSlate = Color(0xFF334155);
  static const parentMist = Color(0xFFF1F5F9);
}

abstract final class PokoOperationToken {
  static const colours = <String, Color>{
    'add': PokoColour.coral,
    'sub': PokoColour.marine,
    'mul': PokoColour.kelp,
    'div': PokoColour.sunfish,
    'wild': PokoColour.violet,
  };

  static const shapes = <String, String>{
    'add': 'roundedSquare',
    'sub': 'circle',
    'mul': 'hexagon',
    'div': 'diamond',
    'wild': 'star',
  };

  static const glyphs = <String, String>{
    'add': '+',
    'sub': '−',
    'mul': '×',
    'div': '÷',
    'wild': '✦',
  };
}

abstract final class PokoSpace {
  static const values = <double>[0, 4, 8, 12, 16, 24, 32, 48, 64];
}

abstract final class PokoRadius {
  static const tile = 8.0;
  static const card = 16.0;
  static const sheet = 24.0;
  static const pill = 999.0;
}

abstract final class PokoTypeSize {
  static const target = 56.0;
  static const tile = 34.0;
  static const preview = 28.0;
  static const h1 = 24.0;
  static const body = 17.0;
  static const caption = 14.0;
}

abstract final class PokoFont {
  static const display = 'Baloo2-ExtraBold';
  static const body = 'Nunito-SemiBold';
  static const bodyBold = 'Nunito-ExtraBold';
  static const dyslexic = 'Lexend-Regular';
}

abstract final class PokoTouch {
  /// Hard floor for every interactive child-zone element. [INV-14]
  static const min = 64.0;
}

abstract final class PokoMotion {
  static const fast = Duration(milliseconds: 120);
  static const base = Duration(milliseconds: 260);
  static const slow = Duration(milliseconds: 450);
  static const reduced = Duration(milliseconds: 100);
  static const maxInputBlock = Duration(milliseconds: 300);
}

ThemeData pokoFoundationTheme() => ThemeData(
  colorScheme: ColorScheme.fromSeed(
    seedColor: PokoColour.tide,
    surface: PokoColour.sand,
  ),
  scaffoldBackgroundColor: PokoColour.sand,
  textTheme: const TextTheme(
    bodyLarge: TextStyle(color: PokoColour.deep, fontSize: PokoTypeSize.body),
  ),
);
